examples/systemcalls/systemcalls.c

```diff
@@ -84,14 +139,46 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     // and may be removed
     command[count] = command[count];
 
+    // The following code is referenced from https://stackoverflow.com/a/13784315/1446624
 
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+    if (fd < 0)
+    {
+        syslog(LOG_ERR, "Failed opening output file %s with an error: %s", outputfile, strerror(errno));
+        return false;
+    }
+    switch (pid = fork()) {
```

Comment:
@krishnasuhagiya29 how could you avoid duplicating logic with the function above?

Reply:
By creating a function doing common operations and calling it from here and from do_exec(). Thanks for pointing out!